src/crypto/signature.js

```diff
@@ -130,6 +144,17 @@ export default {
           util.uint8ArrayToMpi(signature.S)
         ]);
       }
+      case enums.publicKey.cmac: {
+        const algo = key_params[0];
+        const key = key_params[2];
+        if (algo.getName().substr(0, 3) !== 'aes') {
+          throw new Error('CMAC supports only AES cipher');
+        }
+        const cmac = await CMAC(key);
+        const mac = await cmac(hashed);
+
+        return util.uint8ArrayToMpi(mac);
```

Comment:
I'm abusing the notion of MPI here to avoid a big refactor of the signing code. The types I added could be used here but since the whole type wrapper object concept is going away, I didn't feel it was worth it for this. Open to feedback on this

Reply:
Yeah, that's fine with me :+1: